nixos/modules/security/pam.nix

```diff
@@ -1048,6 +1057,8 @@ let
                       modulePath = "${pkgs.google-authenticator}/lib/security/pam_google_authenticator.so";
                       settings = {
                         no_increment_hotp = true;
+                        forward_pass = cfg.forwardPass;
+                        nullok = cfg.allowNullPassword;
```

Comment:
`allowNullPassword` controls `nullok` for `pam_unix`, but `nullok` has a different meaning for `pam_google_authenticator`. This should be a new option, under the `security.pam.services.<name>.googleAuthenticator` option. I suggest a descriptive option name like `allowWithoutOtp` or `otpOptional` or similar.

Reply:
Fixed